spec.html

```diff
@@ -34132,7 +35212,10 @@ <h1>TypedArraySpeciesCreate ( _exemplar_, _argumentList_ )</h1>
           1. Assert: _exemplar_ is an Object that has a [[TypedArrayName]] internal slot.
           1. Let _defaultConstructor_ be the intrinsic object listed in column one of <emu-xref href="#table-49"></emu-xref> for _exemplar_.[[TypedArrayName]].
           1. Let _constructor_ be ? SpeciesConstructor(_exemplar_, _defaultConstructor_).
-          1. Return ? TypedArrayCreate(_constructor_, _argumentList_).
+          1. Let _result_ be ? TypedArrayCreate(_constructor_, _argumentList_).
+          1. Assert: _result_ has a [[TypedArrayName]] internal slot.
+          1. If _result_.[[TypedArrayName]] contains the substring `"Big"` and _exemplar_.[[TypedArrayName]] does not contain the substring `"Big"`, or vice versa, throw a *TypeError* exception.
```

Comment:
and here - at the very least, let's make an abstract operation instead of repeating this check multiple times